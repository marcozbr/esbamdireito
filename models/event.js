import database from 'infra/database.js';
import validator from 'models/validator.js';

async function create(object, options = {}) {
  object = validateObject(object);

  const query = {
    text: `INSERT INTO events (type, originator_user_id, originator_ip, metadata)
               VALUES($1, $2, $3, $4) RETURNING *;`,
    values: [object.type, object.originatorUserId, object.originatorIp, object.metadata],
  };

  const results = await database.query(query, options);
  return results.rows[0];
}

function validateObject(object) {
  const cleanObject = validator(object, {
    event: 'required',
  });

  return cleanObject;
}

export default Object.freeze({
  create,
});
