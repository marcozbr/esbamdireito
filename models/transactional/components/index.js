export { DefaultLayout, DefaultLayoutText } from './default-layout';
export { Link, LinkText } from './link';
export { Text } from './text';
