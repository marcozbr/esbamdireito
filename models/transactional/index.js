import { render } from '@react-email/render';

import { ActivationEmailHtml, ActivationEmailText } from './emails/activation';
import { ConfirmationEmailHtml, ConfirmationEmailText } from './emails/confirmation';
import { NotificationEmailHtml, NotificationEmailText } from './emails/notification';
import { RecoveryEmailHtml, RecoveryEmailText } from './emails/recovery';

export const ActivationEmail = (props) => ({
  html: render(ActivationEmailHtml(props)),
  text: ActivationEmailText(props),
});

export const ConfirmationEmail = (props) => ({
  html: render(ConfirmationEmailHtml(props)),
  text: ConfirmationEmailText(props),
});

export const NotificationEmail = (props) => ({
  html: render(NotificationEmailHtml(props)),
  text: NotificationEmailText(props),
});

export const RecoveryEmail = (props) => ({
  html: render(RecoveryEmailHtml(props)),
  text: RecoveryEmailText(props),
});
