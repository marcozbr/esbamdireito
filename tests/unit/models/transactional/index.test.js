import { ActivationEmail, ConfirmationEmail, NotificationEmail, RecoveryEmail } from 'models/transactional';

describe('Transactional model', () => {
  describe('Email layouts', () => {
    it('Activation', () => {
      const username = 'User Test';
      const activationLink = 'https://tabnews.com.br/cadastro/ativar/TOKEN_ID';

      const { html, text } = ActivationEmail({ username, activationLink });

      expect(html).toMatchSnapshot();
      expect(text).toMatchSnapshot();
    });

    it('Confirmation', () => {
      const username = 'User Test';
      const confirmationLink = 'https://tabnews.com.br/perfil/confirmar-email/TOKEN_ID';

      const { html, text } = ConfirmationEmail({ username, confirmationLink });

      expect(html).toMatchSnapshot();
      expect(text).toMatchSnapshot();
    });

    it('Notification', () => {
      const username = 'User Test';
      const bodyReplyLine = '"User2" respondeu à sua publicação "Título publicação".';
      const contentLink = 'https://tabnews.com.br/user2/titulo-publicacao';

      const { html, text } = NotificationEmail({ username, bodyReplyLine, contentLink });

      expect(html).toMatchSnapshot();
      expect(text).toMatchSnapshot();
    });

    it('Recovery', () => {
      const username = 'User Test';
      const recoveryLink = 'https://tabnews.com.br/perfil/confirmar-email/TOKEN_ID';

      const { html, text } = RecoveryEmail({ username, recoveryLink });

      expect(html).toMatchSnapshot();
      expect(text).toMatchSnapshot();
    });
  });
});
