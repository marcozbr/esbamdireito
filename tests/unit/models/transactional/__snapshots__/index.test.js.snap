// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Transactional model > Email layouts > Activation 1`] = `"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html dir="ltr" lang="en"><head><meta content="text/html; charset=UTF-8" http-equiv="Content-Type"/><meta name="x-apple-disable-message-reformatting"/></head><div style="display:none;overflow:hidden;line-height:1px;opacity:0;max-height:0;max-width:0">Ative seu cadastro no TabNews<div> ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿</div></div><body style="background-color:#ffffff;font-family:-apple-system, BlinkMacSystemFont, &#x27;Segoe UI&#x27;, &#x27;Roboto&#x27;, &#x27;Oxygen&#x27;, &#x27;Ubuntu&#x27;, &#x27;Cantarell&#x27;, &#x27;Fira Sans&#x27;, &#x27;Droid Sans&#x27;, &#x27;Helvetica Neue&#x27;, sans-serif"><table align="center" width="100%" border="0" cellPadding="0" cellSpacing="0" role="presentation" style="max-width:37.5em;padding-left:12px;padding-right:12px;margin:0 auto"><tbody><tr style="width:100%"><td><h1 style="color:#333;font-size:24px;font-weight:bold;margin:40px 0;padding:0">Olá, User Test!</h1><a href="https://tabnews.com.br/cadastro/ativar/TOKEN_ID" style="color:#2754C5;text-decoration:underline;font-size:14px;display:block" target="_blank">Clique aqui para ativar seu cadastro no TabNews.</a><div><p style="font-size:14px;line-height:24px;margin:24px 0;color:#333">Se você não conseguir clicar no link, copie e cole o endereço abaixo no seu navegador:</p><code style="background-color:#f3f3f3;color:#333;display:block;font-size:14px;padding:12px;border-radius:8px;word-break:break-all">https://tabnews.com.br/cadastro/ativar/TOKEN_ID</code></div><p style="font-size:14px;line-height:24px;margin:24px 0;color:#333">Caso você não tenha feito esta requisição, ignore esse email.</p><p style="font-size:12px;line-height:22px;margin:16px 0;color:#898989;margin-top:24px;margin-bottom:4px">Atenciosamente, <br/>Equipe TabNews <br/>Rua Antônio da Veiga, 495, Blumenau, SC, 89012-500</p><img alt="TabNews" height="32" src="https://www.tabnews.com.br/favicon.png" style="display:block;outline:none;border:none;text-decoration:none" width="32"/></td></tr></tbody></table></body></html>"`;

exports[`Transactional model > Email layouts > Activation 2`] = `
"Olá, User Test!

Clique no link abaixo para ativar seu cadastro no TabNews:

https://tabnews.com.br/cadastro/ativar/TOKEN_ID

Caso você não tenha feito esta requisição, ignore esse email.

Atenciosamente, 
Equipe TabNews 
Rua Antônio da Veiga, 495, Blumenau, SC, 89012-500"
`;

exports[`Transactional model > Email layouts > Confirmation 1`] = `"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html dir="ltr" lang="en"><head><meta content="text/html; charset=UTF-8" http-equiv="Content-Type"/><meta name="x-apple-disable-message-reformatting"/></head><div style="display:none;overflow:hidden;line-height:1px;opacity:0;max-height:0;max-width:0">Alteração de email no TabNews<div> ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿</div></div><body style="background-color:#ffffff;font-family:-apple-system, BlinkMacSystemFont, &#x27;Segoe UI&#x27;, &#x27;Roboto&#x27;, &#x27;Oxygen&#x27;, &#x27;Ubuntu&#x27;, &#x27;Cantarell&#x27;, &#x27;Fira Sans&#x27;, &#x27;Droid Sans&#x27;, &#x27;Helvetica Neue&#x27;, sans-serif"><table align="center" width="100%" border="0" cellPadding="0" cellSpacing="0" role="presentation" style="max-width:37.5em;padding-left:12px;padding-right:12px;margin:0 auto"><tbody><tr style="width:100%"><td><h1 style="color:#333;font-size:24px;font-weight:bold;margin:40px 0;padding:0">Olá, User Test!</h1><p style="font-size:14px;line-height:24px;margin:24px 0;color:#333">Uma alteração de email foi solicitada.</p><a href="https://tabnews.com.br/perfil/confirmar-email/TOKEN_ID" style="color:#2754C5;text-decoration:underline;font-size:14px;display:block" target="_blank">Clique aqui para confirmar esta alteração.</a><div><p style="font-size:14px;line-height:24px;margin:24px 0;color:#333">Se você não conseguir clicar no link, copie e cole o endereço abaixo no seu navegador:</p><code style="background-color:#f3f3f3;color:#333;display:block;font-size:14px;padding:12px;border-radius:8px;word-break:break-all">https://tabnews.com.br/perfil/confirmar-email/TOKEN_ID</code></div><p style="font-size:14px;line-height:24px;margin:24px 0;color:#333">Caso você não tenha feito esta requisição, ignore esse email.</p><p style="font-size:12px;line-height:22px;margin:16px 0;color:#898989;margin-top:24px;margin-bottom:4px">Atenciosamente, <br/>Equipe TabNews <br/>Rua Antônio da Veiga, 495, Blumenau, SC, 89012-500</p><img alt="TabNews" height="32" src="https://www.tabnews.com.br/favicon.png" style="display:block;outline:none;border:none;text-decoration:none" width="32"/></td></tr></tbody></table></body></html>"`;

exports[`Transactional model > Email layouts > Confirmation 2`] = `
"Olá, User Test!

Uma alteração de email foi solicitada.

Clique no link abaixo para confirmar esta alteração:

https://tabnews.com.br/perfil/confirmar-email/TOKEN_ID
  
Caso você não tenha feito esta requisição, ignore esse email.

Atenciosamente, 
Equipe TabNews 
Rua Antônio da Veiga, 495, Blumenau, SC, 89012-500"
`;

exports[`Transactional model > Email layouts > Notification 1`] = `"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html dir="ltr" lang="en"><head><meta content="text/html; charset=UTF-8" http-equiv="Content-Type"/><meta name="x-apple-disable-message-reformatting"/></head><div style="display:none;overflow:hidden;line-height:1px;opacity:0;max-height:0;max-width:0">Nova resposta no TabNews<div> ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿</div></div><body style="background-color:#ffffff;font-family:-apple-system, BlinkMacSystemFont, &#x27;Segoe UI&#x27;, &#x27;Roboto&#x27;, &#x27;Oxygen&#x27;, &#x27;Ubuntu&#x27;, &#x27;Cantarell&#x27;, &#x27;Fira Sans&#x27;, &#x27;Droid Sans&#x27;, &#x27;Helvetica Neue&#x27;, sans-serif"><table align="center" width="100%" border="0" cellPadding="0" cellSpacing="0" role="presentation" style="max-width:37.5em;padding-left:12px;padding-right:12px;margin:0 auto"><tbody><tr style="width:100%"><td><h1 style="color:#333;font-size:24px;font-weight:bold;margin:40px 0;padding:0">Olá, User Test!</h1><p style="font-size:14px;line-height:24px;margin:24px 0;color:#333">&quot;User2&quot; respondeu à sua publicação &quot;Título publicação&quot;.</p><a href="https://tabnews.com.br/user2/titulo-publicacao" style="color:#2754C5;text-decoration:underline;font-size:14px;display:block" target="_blank">Clique aqui para ler a resposta.</a><div><p style="font-size:14px;line-height:24px;margin:24px 0;color:#333">Se você não conseguir clicar no link, copie e cole o endereço abaixo no seu navegador:</p><code style="background-color:#f3f3f3;color:#333;display:block;font-size:14px;padding:12px;border-radius:8px;word-break:break-all">https://tabnews.com.br/user2/titulo-publicacao</code></div><p style="font-size:12px;line-height:22px;margin:16px 0;color:#898989;margin-top:24px;margin-bottom:4px">Atenciosamente, <br/>Equipe TabNews <br/>Rua Antônio da Veiga, 495, Blumenau, SC, 89012-500</p><img alt="TabNews" height="32" src="https://www.tabnews.com.br/favicon.png" style="display:block;outline:none;border:none;text-decoration:none" width="32"/></td></tr></tbody></table></body></html>"`;

exports[`Transactional model > Email layouts > Notification 2`] = `
"Olá, User Test!

"User2" respondeu à sua publicação "Título publicação". Para ler a resposta, utilize o link abaixo:

https://tabnews.com.br/user2/titulo-publicacao

Atenciosamente, 
Equipe TabNews 
Rua Antônio da Veiga, 495, Blumenau, SC, 89012-500"
`;

exports[`Transactional model > Email layouts > Recovery 1`] = `"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html dir="ltr" lang="en"><head><meta content="text/html; charset=UTF-8" http-equiv="Content-Type"/><meta name="x-apple-disable-message-reformatting"/></head><div style="display:none;overflow:hidden;line-height:1px;opacity:0;max-height:0;max-width:0">Recuperação de senha no TabNews<div> ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿</div></div><body style="background-color:#ffffff;font-family:-apple-system, BlinkMacSystemFont, &#x27;Segoe UI&#x27;, &#x27;Roboto&#x27;, &#x27;Oxygen&#x27;, &#x27;Ubuntu&#x27;, &#x27;Cantarell&#x27;, &#x27;Fira Sans&#x27;, &#x27;Droid Sans&#x27;, &#x27;Helvetica Neue&#x27;, sans-serif"><table align="center" width="100%" border="0" cellPadding="0" cellSpacing="0" role="presentation" style="max-width:37.5em;padding-left:12px;padding-right:12px;margin:0 auto"><tbody><tr style="width:100%"><td><h1 style="color:#333;font-size:24px;font-weight:bold;margin:40px 0;padding:0">Olá, User Test!</h1><p style="font-size:14px;line-height:24px;margin:24px 0;color:#333">Uma recuperação de senha foi solicitada.</p><a href="https://tabnews.com.br/perfil/confirmar-email/TOKEN_ID" style="color:#2754C5;text-decoration:underline;font-size:14px;display:block" target="_blank">Clique aqui para definir uma nova senha.</a><div><p style="font-size:14px;line-height:24px;margin:24px 0;color:#333">Se você não conseguir clicar no link, copie e cole o endereço abaixo no seu navegador:</p><code style="background-color:#f3f3f3;color:#333;display:block;font-size:14px;padding:12px;border-radius:8px;word-break:break-all">https://tabnews.com.br/perfil/confirmar-email/TOKEN_ID</code></div><p style="font-size:14px;line-height:24px;margin:24px 0;color:#333">Caso você não tenha feito esta requisição, ignore esse email.</p><p style="font-size:12px;line-height:22px;margin:16px 0;color:#898989;margin-top:24px;margin-bottom:4px">Atenciosamente, <br/>Equipe TabNews <br/>Rua Antônio da Veiga, 495, Blumenau, SC, 89012-500</p><img alt="TabNews" height="32" src="https://www.tabnews.com.br/favicon.png" style="display:block;outline:none;border:none;text-decoration:none" width="32"/></td></tr></tbody></table></body></html>"`;

exports[`Transactional model > Email layouts > Recovery 2`] = `
"Olá, User Test!

Uma recuperação de senha foi solicitada. Caso você não tenha feito a solicitação, ignore esse email.

Caso você tenha feito essa solicitação, clique no link abaixo para definir uma nova senha:

https://tabnews.com.br/perfil/confirmar-email/TOKEN_ID

Atenciosamente, 
Equipe TabNews 
Rua Antônio da Veiga, 495, Blumenau, SC, 89012-500"
`;
